colcon_library_path/environment/library_path.py

```diff
@@ -34,22 +34,30 @@ def create_environment_hooks(self, prefix_path, pkg_name):  # noqa: D102
             ['hook_name', 'subdirectory', 'extension', 'environment_variable'])
 
         if sys.platform == 'win32':
-            library_descriptor = LibraryDescriptor(
-                'path_dll', 'bin', 'dll', 'PATH')
+            library_descriptors = [
+                LibraryDescriptor('path_dll', 'bin', 'dll', 'PATH')
+            ]
         elif platform.system() == 'Darwin':
-            library_descriptor = LibraryDescriptor(
-                'dyld_library_path', 'lib', 'dylib', 'DYLD_LIBRARY_PATH')
+            library_descriptors = [
+                LibraryDescriptor(
+                    'dyld_library_path', 'lib', 'dylib', 'DYLD_LIBRARY_PATH')]
         else:
-            library_descriptor = LibraryDescriptor(
-                'ld_library_path', 'lib', 'so', 'LD_LIBRARY_PATH')
-
-        library_path = prefix_path / library_descriptor.subdirectory
-        logger.log(1, "checking '%s'" % library_path)
-
-        if not any(library_path.glob('*.' + library_descriptor.extension)):
-            return []
-        return create_environment_hook(
-            library_descriptor.hook_name,
-            prefix_path, pkg_name,
-            library_descriptor.environment_variable,
-            library_descriptor.subdirectory, mode='prepend')
+            library_descriptors = [
+                LibraryDescriptor(
+                    'ld_library_path', 'lib', 'so', 'LD_LIBRARY_PATH'),
+                LibraryDescriptor(
+                    'ld_library_path', 'lib64', 'so', 'LD_LIBRARY_PATH')]
```

Comment:
Does it make sense to hard code `lib64` here? Potentially there could be others like `lib32` or `libx32`.

Two possible options would be to:
* list all possible ones
* use some kind of detection which one to be added (since the build could easily be a cross compilation this might be impossible to get right)

Reply:
Have you tried the patch with both paths containing libraries? I would assume that both environment hooks end up with the same name (based on the first argument) and therefore overwriting each other.

Reply:
> Does it make sense to hard code lib64 here?

I was also wondering that. This was initially because CMake's `GNUInstallDirs.cmake` only uses `lib` or `lib64`. When cross-compiling, it uses `lib`. But a package could in theory use anything.

The [GNU standards](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html) that `GNUInstallDirs.cmake` refers to don't say anything about this, just that it normally should just be `lib`.

The Filesystem Hierarchy Standard [does mention `lib<qual>`](http://www.pathname.com/fhs/pub/fhs-2.3.html#USRLIBLTQUALGTALTERNATEFORMATLIBRARI), without further limits.

Based on that I think globbing for `lib*` makes sense, what do you reckon?

Reply:
> Have you tried the patch with both paths containing libraries?

No I haven't, I will make sure they have unique names.

Reply:
I have implemented the globbing, let me know what you think!

Reply:
Thanks for iterating on it. I added a minor change to make it a bit more readable.